presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -101,6 +102,9 @@
     // for describe input and describe output
     private final boolean isDescribe;
 
+    // for recursive view detection
+    private final HashSet<Table> tablesForView = new HashSet<>();
```

Comment:
Declare variables using the interface type, not the concrete implementation. So this should be `Set` instead of `HashSet`.